erpnext/setup/doctype/naming_series/naming_series.py

```diff
@@ -34,7 +34,14 @@ def get_transactions(self, arg=None):
 			if options:
 				prefixes = prefixes + "\n" + options
 		prefixes.replace("\n\n", "\n")
-		prefixes = "\n".join(sorted(prefixes.split("\n")))
+		prefixes = prefixes.split("\n")
+
+		custom_prefixes = frappe.get_all('DocType', fields=["autoname"],
+			filters={"name": ('not in', doctypes), "autoname":('like', '%.#%')})
+		if custom_prefixes:
+			prefixes = prefixes + [d.autoname.rsplit('.')[0] for d in custom_prefixes]
```

Comment:
`d.autoname.rsplit('.')[0]` will fail if autoname is like `SO.customer_name.####` or `SO.YY.####`

Reply:
Fixed!